contrib/core-contract-tests/tests/pox-4/pox-4.prop.test.ts

```diff
@@ -0,0 +1,434 @@
+          const expected = 0;
+
+          // Act
+          const { result: actual } = simnet.callReadOnlyFn(
+            POX_4,
+            "get-total-ustx-stacked",
+            [Cl.uintCV(reward_cycle)],
+            caller
+          );
+
+          // Assert
+          assert(isClarityType(actual, ClarityType.UInt));
+          expect(actual).toBeUint(expected);
+        }
+      )
+    );
+  });
+
+  it("should return none get-reward-set-pox-address", () => {
+    fc.assert(
+      fc.property(
+        fc.constantFrom(...simnet.getAccounts().values()),
+        fc.nat(),
+        fc.nat(),
+        (caller, index, reward_cycle) => {
+          // Arrange
+
+          // Act
+          const { result: actual } = simnet.callReadOnlyFn(
+            POX_4,
+            "get-reward-set-pox-address",
+            [Cl.uintCV(index), Cl.uintCV(reward_cycle)],
+            caller
+          );
+
+          // Assert
+          assert(isClarityType(actual, ClarityType.OptionalNone));
+          expect(actual).toBeNone();
+        }
+      )
+    );
+  });
+
+  it("should return correct get-stacking-minimum", () => {
+    fc.assert(
+      fc.property(
+        fc.constantFrom(...simnet.getAccounts().values()),
+        (caller) => {
+          // Arrange
+          const testnet_stacking_threshold_25 = 8000;
+
+          const { result: pox_4_info } = simnet.callReadOnlyFn(
+            POX_4,
+            GET_POX_INFO,
+            [],
+            caller
+          );
+          assert(isClarityType(pox_4_info, ClarityType.ResponseOk));
+          assert(isClarityType(pox_4_info.value, ClarityType.Tuple));
+
+          const stx_liq_supply =
+            pox_4_info.value.data["total-liquid-supply-ustx"];
+
+          assert(isClarityType(stx_liq_supply, ClarityType.UInt));
+          const expected = Math.floor(
+            Number(stx_liq_supply.value) / testnet_stacking_threshold_25
+          );
+
+          // Act
+          const { result: actual } = simnet.callReadOnlyFn(
+            POX_4,
+            "get-stacking-minimum",
+            [],
+            caller
+          );
+
+          // Assert
+          assert(isClarityType(actual, ClarityType.UInt));
+          expect(actual).toBeUint(expected);
+        }
+      )
+    );
+  });
+
+  it("should return correct check-pox-addr-version", () => {
+    fc.assert(
+      fc.property(
+        fc.constantFrom(...simnet.getAccounts().values()),
+        fc.nat({ max: 255 }),
+        (caller, version) => {
+          // Arrange
+          const expected = version > 6 ? false : true;
+
+          // Act
+          let { result: actual } = simnet.callReadOnlyFn(
+            POX_4,
+            "check-pox-addr-version",
+            [Cl.bufferCV(Uint8Array.from([version]))],
+            caller
+          );
+
+          // Assert
+          assert(
+            isClarityType(
+              actual,
+              expected ? ClarityType.BoolTrue : ClarityType.BoolFalse
```

Comment:
I'm a bit more unfamiliar w/ Vitest than I'd like but this ternary throws me off? If we set "expected" in line 267 why is there a ternary in the asserts statement?

Couldn't that return a false positive / always succeed? Or am I misunderstanding the purpose here?

Reply:
That assert basically checks that the type of the `actual` result is the right one. If the randomly generated version is greater than 6, then the `actual`'s clarity type should be false. Else, it should be true. I'll split this into 2 separated tests to avoid the ternary inside the assertion.